### ISSUE TITLE
The change of value attribute to null in DatePicker

### ISSUE THREAD

**Issue by MichalWadowski** [mlaursen/react-md#409]
When we create the component DatePicker, which has value other than null and next we change the value of the component to null, then the following errors occur:

```
Failed prop type: The prop `calendarDate` is marked as required in `DatePicker`,
but its value is `null`.
```

```
Failed prop type: The prop `calendarTempDate` is
marked as required in `DatePicker`, but its value is `null`
```

When the controll is initialized with the value=null, then the error does not occur.

The example: https://codepen.io/MichalWadowski/pen/jwbaqz

This issue may be related to #384

**Comment by chase** [comment]
I setup this little patched wrapper for when I use DatePicker to get around this issue:
``` javascript
import React, { Component } from 'react';
import DatePicker from 'react-md/lib/Pickers/DatePickerContainer';
import isDateEqual from 'react-md/lib/utils/DateUtils/isDateEqual';

class DatePickerWrapper class Component {
    constructor(props) {
        super(props);
        this.injectContainer = this.injectContainer.bind(this);
    }

    injectContainer(container) {
        if (!container) return;
        function componentWillReceiveProps(nextProps) {
            let { value } = nextProps;
            const { defaultValue } = this.props;
            const { minDate, maxDate, initialCalendarDate } = nextProps;
            const minEqual = isDateEqual(this.props.minDate, minDate);
            const maxEqual = isDateEqual(this.props.maxDate, maxDate);
            if (this.props.value !== value || !minEqual || !maxEqual) {
                let { calendarDate } = this.state;
                if (!!value) {
                    calendarDate = typeof value === 'string' ? new Date(value) : value;
                } else if (defaultValue) {
                    calendarDate = typeof defaultValue === 'string' ? new Date(defaultValue) : defaultValue;
                    value = typeof defaultValue === 'string'
                        ? defaultValue
                        : DateTimeFormat(locales, formatOptions).format(defaultValue);
                } else {
                    calendarDate = new Date();
                }

                calendarDate = this._validateDateRange(calendarDate, minDate, maxDate);

                let calendarTempDate = calendarDate;
                if (typeof initialCalendarDate !== 'undefined' && !value && !defaultValue) {
                    calendarTempDate = typeof initialCalendarDate === 'string'
                        ? new Date(initialCalendarDate)
                        : initialCalendarDate;
                    calendarDate = calendarTempDate;
                } else if (calendarTempDate === null) {
                    calendarTempDate = new Date();
                    calendarDate = new Date();
                }

                if (!isDateEqual(this.state.calendarDate, calendarDate)) {
                    this.setState({ calendarDate, calendarTempDate });
                }
            }
        }
        this.dateContainer.componentWillReceiveProps = componentWillReceiveProps.bind(this.dateContainer);
    }
    render() (
        <DatePicker
            {...this.props}
            ref={this.injectContainer}
        />
    );
}
```

This could be made into a proper patch, but essentially I took what was in the constructor and replaced the similar parts of `componentWillReceiveProps` in DatePickerContainer. I'm sure the correct thing to do would be to pull the common behavior out into a function and call it in both.

**Comment by mlaursen** [comment]
Hmm. Doesn't look like it linked correctly, but this should be fixed in 8b4b553.
